include/ldsCtrlEst_h/lds_mpcctrl.h

```diff
@@ -0,0 +1,533 @@
+//===-- ldsCtrlEst_h/lds_gaussian_mpcctrl.h - MPC Controller ------*- C++
+//-*-===//
+//
+// Copyright 2024 Chia-Chien Hung and Kyle Johnsen
+// Copyright 2024 Georgia Institute of Technology
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file defines the type for model predictive Model Predictive
+/// Control (MPC) on linear system dynamics by converting the MPC optimization
+/// problem to a quadratic cost problem. The new problem is optimized using the
+/// Operator Splitting Quadratic Program (OSQP).
+///
+/// \brief MPC Controller
+//===----------------------------------------------------------------------===//
+
+#ifndef LDSCTRLEST_LDS_MPCCTRL_H
+#define LDSCTRLEST_LDS_MPCCTRL_H
+
+// namespace
+#include "lds.h"
+// system type
+#include "lds_sys.h"
+
+// osqp
+#include "osqp.h"
+#include "osqp_api_constants.h"
+
+namespace lds {
+
+template <typename System>
+class MpcController {
```

Comment:
Is there any way for MpcController to actually implement the Controller interface in lds_ctrl.h? I think probably not since there's so much different stuff (OSQP, reference over the horizon, etc.), but just wanted to confirm

Reply:
I thought it might be quite difficult or kind of strange to implement it in that way, but I feel like I need to understand MPC better to really know if something like that is possible. I will let you know once I learn more.